pkg/gofr/container/mock_container.go

```diff
@@ -75,7 +114,114 @@ func NewMockContainer(t *testing.T) (*Container, Mocks) {
 	return container, mocks
 }
 
-type MockPubSub struct {
+func (m sqlMockDB) Select(_ context.Context, _ interface{}, query string, args ...interface{}) {
+	if m.queryWithArgs == nil || len(m.queryWithArgs) == 0 {
+		m.logger.Fatalf("Did not expect any calls for Select with query: %s", query)
+	}
+
+	lastIndex := len(m.queryWithArgs) - 1
+	expectedText := m.queryWithArgs[lastIndex].queryText
+	expectedArgs := m.queryWithArgs[lastIndex].arguments
+
+	if expectedText != query {
+		m.logger.Fatalf("expected query: %s, actual query: %s", query, expectedText)
+	}
+
+	if len(args) != len(expectedArgs) {
+		m.logger.Fatalf("expected %d args, actual %d", len(expectedArgs), len(args))
+	}
+
+	for i := range args {
+		if args[i] != expectedArgs[i] {
+			m.logger.Fatalf("expected arg %d, actual arg %d", args[i], expectedArgs[i])
+		}
+	}
+
+	m.queryWithArgs = m.queryWithArgs[:lastIndex]
+}
+
+func (m sqlMockDB) HealthCheck() *datasource.Health {
+	if m.expectedHealthCheck == nil || len(m.expectedHealthCheck) == 0 {
+		m.logger.Fatal("Did not expect any mock calls for HealthCheck")
+	}
+
```

Comment:
I'm surprised this file and a few others do not use testify but plain testing

I mean here, I would have expected something like this

```suggestion
	require.NotEmpty(m.expectedHealthCheck, "Did not expect any mock calls for HealthCheck")
```

Reply:
I don't pass the t *Testing.T in the functions that is required for require and assert functions.